scipy/signal/tests/test_signaltools.py

```diff
@@ -512,18 +568,20 @@ def test_complex_axes(self, axes):
                                       [1, -2],
                                       [-2, -1],
                                       [-1, -2]])
-    def test_2d_real_same(self, axes):
-        a = array([[1, 2, 3],
+    def test_2d_real_same(self, axes, xp):
+        a = xp.asarray([[1, 2, 3],
                    [4, 5, 6]])
-        expected = array([[1, 4, 10, 12, 9],
+        expected = xp.asarray([[1, 4, 10, 12, 9],
                           [8, 26, 56, 54, 36],
                           [16, 40, 73, 60, 36]])
 
         if axes == '':
             out = fftconvolve(a, a)
         else:
+            if isinstance(axes, list):
+                axes = tuple(axes)
             out = fftconvolve(a, a, axes=axes)
-        assert_array_almost_equal(out, expected)
+        xp_assert_close(out, expected, atol=1.5e-6, check_dtype=False)
```

Comment:
For the tests that are skipped elsewhere for dtype reasons could we not turn off the check like here for the troublesome backends?

Reply:
1. apparently, tests still pass with a tighter atol; 2. `check_dtype=False` is not needed if input/expected arrays are floats not ints.
There's no inherent reason to use integer arrays, so changed it here and simplified the assertion.